### DISCUSSION TITLE
Does rueidis support fetching multiple keys across different slots without client side cacheing enabled?

### DISCUSSION THREAD

**Discussion by lchalabi** [redis/rueidis#177] [Q&A] [answered]
Does rueidis support fetching multiple keys across different slots with MGET without client side cacheing enabled?   I see the helper `rueidis.MGetCache` which leverages `cmds.MGets` to group keys by their slot and return multi MGET commands.  However, it's not clear to me whether this is supported without client side cacheing.   Thank you!

**Comment by rueian** [comment, marked as answer]
Hi @lchalabi,

`rueidis.MGet` helper has been added to v0.0.94. You can now use it to fetch multiple keys across different slots with MGET without client side caching.

**Reply by lchalabi** [reply]
Great! is there a similar utility to use `MSET` across different slots?  Thank you.

**Reply by rueian** [reply]
Hi @lchalabi,

Currently, there is no similar utilities for `MSET` and `MSETNX`.

Alternatively, I would recommend using normal `SET` instead. It is actually more powerful than `MSET` with the ability to setting options such as `EX`, `PX`, `NX` and `XX`.

You can either simply use multiple goroutines to send `SET` commends with `client.Do()` at the same time or pre-construct them first and then use `client.DoMulti()` at once, for example:

```go
cmds := make(rueidis.Commands, 0, 10)
for i := 0; i < 10; i++ {
    cmds = append(cmds, client.B().Set().Key("key").Value("value").Build())
}
for _, resp := range client.DoMulti(ctx, cmds...) {
    if err := resp.Error(); err != nil {
        panic(err)
    }
}
```

**Reply by lchalabi** [reply]
Thank you @rueian!

**Reply by rueian** [reply]
Hi @lchalabi, we have MSet and MSetNX helpers now!

**Reply by lchalabi** [reply]
Great! would this allow us to better optimize the distribution of data across slots?  Perhaps optimizing to store data in the fewer slots to reduce the number of MGETs needed on the read path?  We're currently seeing elevated CPU due the high number of MGETs needed to read batched data from our cluster.

**Reply by rueian** [reply]
To reduce slots required by your keys, you need to specify slot mapping with `{xxx}` in your keys.
